c10/core/TensorImpl.h

```diff
@@ -1636,16 +1643,12 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
   // occurs in THPVariable_clear in torch/csrc/autograd/python_variable.cpp
   PyObject* pyobj_ = nullptr;
 
-  // We could save a word or two by combining the SmallVector structs,
-  // since their size is redundant, and if we need to overflow the buffer space
-  // we could keep the two pointers together. However, that would require
-  // implementing another struct from scratch, so only do this if we're desperate.
-  SmallVector<int64_t,5> sizes_;
-  SmallVector<int64_t,5> strides_;
+  c10::impl::SizesAndStrides sizes_and_strides_;
 
   int64_t storage_offset_ = 0;
   // If sizes and strides are empty, the numel is 1!!  However, most of the
   // time, we will immediately set sizes to {0} and reset numel to 0.
+  // XXX: what is this talking about exactly?
```

Comment:
If you have a tensor where `x.sizes() == {}`, then (somewhat unintuitively) its `x.numel() == 1`. But if `x.sizes() == {0}` then `x.numel() == 0`.

Reply:
so what can't we do in the default initializers exactly? I think before and after this diff we already had 1-element sizes and strides after construction.